internal/parser/scanner/scanner.go

```diff
@@ -283,18 +228,23 @@ func (s *scanner) peekString(str string) bool {
 }
 
 // createToken creates a token with given parameters
-func createToken(line, col, start, pos int, t token.Type, value string, s *scanner) token.Token {
-	token := token.New(line, col, start, pos-start, t, value)
-	s.start = pos
-	return token
+func (s *scanner) createToken(t token.Type) token.Token {
+	tk := token.New(s.line, s.col, s.start, s.pos-s.start, t, string(s.input[s.start:s.pos]))
+	s.start = s.pos
+	return tk
 }
 
-// seekNext returns the position of the end of a keyword.
+// seekNextPos returns the position of the end of a keyword.
 // It takes the start position of the keyword.
-func (s *scanner) seekNext(start int) int {
-	start++
-	for s.input[start] != ' ' {
+func (s *scanner) seekNextPos(start int) int {
```

Comment:
What about `seekNextTokenStart` or `seekTokenEnd`?

Reply:
Ill change it to `seekTokenEnd` :)